src/language_server_protocol.rs

```diff
@@ -23,14 +25,15 @@ impl LanguageClient {
             })
     }
 
-    pub fn loop_call(&self, rx: &crossbeam::channel::Receiver<Call>) -> Fallible<()> {
+    pub async fn loop_call(&self, rx: &crossbeam::channel::Receiver<Call>) -> Fallible<()> {
         for call in rx.iter() {
             let language_client = LanguageClient {
                 version: self.version.clone(),
                 state_mutex: self.state_mutex.clone(),
                 clients_mutex: self.clients_mutex.clone(), // not sure if useful to clone this
             };
-            thread::spawn(move || {
+
+            tokio::task::spawn_blocking(move || {
```

Comment:
From what I'm reading of the doc, this will block the caller thread, i.e., it won't process the next message until the previous one is completed.

Whereas here what we want is concurrently processing all the requests.

Reply:
Ah yeah you're right. I was missing an `.await` there and I guess that I was also lucky that it didn't block but yeah, `tokio::spawn` is what we want there instead. Fixed!